src/rules/no-descending-specificity/index.js

```diff
@@ -52,17 +52,24 @@ export default function (actual) {
         return
       }
 
+      // Return early if selector contains a matches
+      if (selector.indexOf(":matches(") !== -1) { return }
+
       const priorComparableSelectors = comparisonContext.get(lastNonPseudoSelectorNode)
 
       priorComparableSelectors.forEach(priorEntry => {
-        if (compare(selectorSpecificity, priorEntry.specificity) === -1) {
-          report({
-            ruleName,
-            result,
-            node: rule,
-            message: messages.rejected(selector, priorEntry.selector),
-            index: sourceIndex,
-          })
+        try {
+          if (compare(selectorSpecificity, priorEntry.specificity) === -1) {
+            report({
+              ruleName,
+              result,
+              node: rule,
+              message: messages.rejected(selector, priorEntry.selector),
+              index: sourceIndex,
+            })
+          }
+        } catch (e) {
+          result.warn("cannot parse selector", { selectorNode })
```

Comment:
`no-descending-specificity`, i don't think we need the check for matches here at all.

In case of `selector-max-specificity` we pass selector, but here we always pass in the array

I think I have to revert this. 

@ntwb @davidtheclark @jeddy3 @evilebottnawi